src/visibility.js

```diff
@@ -0,0 +1,173 @@
+        props: {
+          onClose: () => {
+            this._dismissed();
+          }
+        }
+      }
+    ];
+  }
+
+  /**
+   * @static
+   * @public
+   * @returns {boolean} - Whether the plugin is valid.
+   */
+  static isValid(): boolean {
+    return true;
+  }
+
+  /**
+   * @constructor
+   * @param {string} name - The plugin name.
+   * @param {Player} player - The player instance.
+   * @param {Object} config - The plugin config.
+   */
+  constructor(name: string, player: Player, config: Object) {
+    super(name, player, config);
+    this.reset();
+    if (this.config.floating) {
+      this._initFloating();
+    }
+    const options = {
+      threshold: this.config.threshold
+    };
+
+    this.observer = new window.IntersectionObserver(this._handleVisibilityChange.bind(this), options);
+    this.observer.observe(this.appTargetContainer);
+  }
+
+  _initFloating() {
+    const defaultFloatingConfig = {
+      floating: {
+        position: 'bottom-right',
+        size: {
+          height: '225px',
+          width: '400px'
+        },
+        margin: {
+          x: '2px',
+          y: '2px'
+        }
+      }
+    };
+    Utils.Object.mergeDeep(this.config, defaultFloatingConfig, Utils.Object.copyDeep(this.config));
+    this.floatingPoster = Utils.Dom.createElement('div');
+    this.floatingPoster.className = FLOATING_POSTER_CLASS;
+    this.floatingContainer = Utils.Dom.createElement('div');
+    this.floatingContainer.className = FLOATING_CONTAINER_CLASS;
+
+    this.appTargetContainer = Utils.Dom.getElementById(this.player.config.targetId);
+    this.floatingContainer.innerHTML = this.appTargetContainer.innerHTML;
+    this.appTargetContainer.innerHTML = '';
+    Utils.Dom.appendChild(this.appTargetContainer, this.floatingPoster);
+    Utils.Dom.appendChild(this.appTargetContainer, this.floatingContainer);
+    this.config.floating.position.split('-').forEach(side => {
+      Utils.Dom.addClassName(this.floatingContainer, `${FLOATING_ACTIVE_CLASS}-${side}`);
+    });
+  }
+
+  _dismissed() {
+    this.dismissed = true;
+    this.player.pause();
+    this._stopFloating();
+  }
+
+  _stopFloating() {
+    Utils.Dom.removeClassName(this.floatingContainer, FLOATING_ACTIVE_CLASS);
+    Utils.Dom.removeAttribute(this.floatingContainer, 'style');
+  }
+
+  _startFloating() {
+    Utils.Dom.addClassName(this.floatingContainer, FLOATING_ACTIVE_CLASS);
+    Utils.Dom.setStyle(this.floatingContainer, 'height', this.config.floating.size.height);
+    Utils.Dom.setStyle(this.floatingContainer, 'width', this.config.floating.size.width);
+    Utils.Dom.setStyle(this.floatingContainer, 'margin', `${this.config.floating.margin.y} ${this.config.floating.margin.x}`);
+  }
+
+  _handleVisibilityChange(entries: Array<window.IntersectionObserverEntry>) {
+    const playerIsOutOfVisibility = entries[0].intersectionRatio < this.config.threshold;
+    if (this.config.floating && this.everStartedPlaying && !this.dismissed) {
```

Comment:
maybe `dismissed` should be static to be applied for all players on the page

Reply:
We currently only support one floating player in a page. If we want to support more than one then we need to make a bigger change

Reply:
I'm not talking about multiple floating players.
Just think about a user that dismiss the floating player and move to the next player - 
in your impl he will get a floating player again. I think it's wrong behavior